### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by yuvenalmash**
# Made the following changes:
- Add an event listener and resulting functions for the checkboxes
- Implement a function for the "Clear all completed" button
- Store the updated array of items in local storage

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+  const toDoList = JSON.parse(localStorage.getItem('allEntries'));
+  const complete = toDoList.filter((toDo) => toDo.completed === true);
+  const indexes = [];
+  for (let i = 0; i < complete.length; i += 1) {
+    const index = (complete[i].index) - 1;
+    indexes.push(index);
+  }
+
+  for (let i = indexes.length - 1; i >= 0; i -= 1) toDoList.splice(indexes[i], 1);
+
```
Comment (Gambit142) [inline src/index.js]: - [ ] According to the requirements, you should use the `filter` method to remove all completed tasks from your to-do list. Kindly change the method used to do so from `splice` to `filter`. 🙏 👍 

image

File: src/modules/statusUpdate.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const check = document.getElementById(`check${index}`);
+  const description = document.getElementById(`description${index}`);
+  const toDoList = JSON.parse(localStorage.getItem('allEntries'));
+  if (check.checked) {
+    toDoList[index].completed = true;
+    description.style.textDecoration = 'line-through';
+  } else {
+    toDoList[index].completed = false;
+    description.style.textDecoration = 'none';
+  }
```
Comment (Gambit142) [inline src/modules/statusUpdate.js:11]: - [ ] Great work implementing this feature this way. However, you are meant to change the status of tasks by adding a `change` `eventlistener` to the checkbox. Kindly implement this feature. 👍 

image

`checkbox.addEventlistener('change', callbackFunction)`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @yuvenalmash 👋, 

Good job so far!

## Highlights
- All linters checks passed. 👍 
- The README.md file is very professional. 👍 
- Commit message is in the imperative tense. 👍 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### I am making this comment here because I make inline comments to your README.md file.

- You have a great _**README.md**_ file and most commands are accurate. Kindly remove the Microverse logo in the README.md file. If you have your custom logo, you can use it but if you don't, kindly remove that section from the file.

_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by frank1738 — APPROVED** [review]
Hi @yuvenalmash , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Github workflow
✅ No linter error
✅ README is okay
✅ Good PR Description 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**